pkg/registry/controller/strategy.go

```diff
@@ -92,9 +93,31 @@ func (rcStrategy) AllowCreateOnUpdate() bool {
 
 // ValidateUpdate is the default update validation for an end user.
 func (rcStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) field.ErrorList {
-	validationErrorList := validation.ValidateReplicationController(obj.(*api.ReplicationController))
-	updateErrorList := validation.ValidateReplicationControllerUpdate(obj.(*api.ReplicationController), old.(*api.ReplicationController))
-	return append(validationErrorList, updateErrorList...)
+	oldRc := old.(*api.ReplicationController)
+	newRc := obj.(*api.ReplicationController)
+
+	validationErrorList := validation.ValidateReplicationController(newRc)
+	updateErrorList := validation.ValidateReplicationControllerUpdate(newRc, oldRc)
+	errs := append(validationErrorList, updateErrorList...)
+
+	for key, value := range api.NonConvertibleFields(oldRc.Annotations) {
```

Comment:
> Why do this in a loop? Just look for the field you care about, and then also verify that the map length is 1?

If the map has zero length, the loop never runs. These annotations are important for clients so going over them one by one seems like the sanest thing to do.

> This code hasn't been run in any kubernetes cluster yet, so we don't have a backwards compatibility problem. (OpenShift might, but do we have to upstream that?)

Aren't you planning on moving replica sets and replication controllers under the same etcd key at some point? Seems like you would need this code.